V error when defer inside $if cond {} $else {}

```
OS: linux, Ubuntu 20.04.5 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i5-3320M CPU @ 2.60GHz
CC version: cc (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0

getwd: /home/woody/sources/v/vhome/bugs
vmodules: /home/woody/.vmodules
vroot: /home/woody/sources/v/v
vexe: /home/woody/sources/v/v/v
vexe mtime: 2022-09-26 18:56:29
is vroot writable: true
is vmodules writable: true
V full version: V 0.3.1 17b54cd.385acb4

Git version: git version 2.25.1
Git vroot status: weekly.2022.39
.git/config present: true
thirdparty/tcc status: thirdparty-linux-amd64 827f7452
```


**What did you do?**
bug.v:
```
module main

fn main() {
	$if cond ? {
	} $else {
		defer {
		}
	}
}
```
`v bug.v`

**What did you expect to see?**
no error

 
**What did you see instead?**

```
/tmp/v_1000/ssl3.1060091373529785859.tmp.c:12084: error: #else without matching #if
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang
```

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -2093,6 +2093,7 @@ fn (mut g Gen) stmt(node ast.Stmt) {
 }
 
 fn (mut g Gen) write_defer_stmts() {
+	g.indent++
 	for i := g.defer_stmts.len - 1; i >= 0; i-- {
 		defer_stmt := g.defer_stmts[i]
 		g.writeln('// Defer begin')
@@ -2112,6 +2113,7 @@ fn (mut g Gen) write_defer_stmts() {
 		g.writeln('}')
 		g.writeln('// Defer end')
 	}
+	g.indent--
 }
 
 struct SumtypeCastingFn {

--- a/vlib/v/gen/c/comptime.v
+++ b/vlib/v/gen/c/comptime.v
@@ -266,7 +266,12 @@ fn (mut g Gen) comptime_if(node ast.IfExpr) {
 			g.writeln('')
 		}
 		expr_str := g.out.last_n(g.out.len - start_pos).trim_space()
-		g.defer_ifdef = expr_str
+		if expr_str != '' {
+			if g.defer_ifdef != '' {
+				g.defer_ifdef += '\r\n' + '\t'.repeat(g.indent + 1)
+			}
+			g.defer_ifdef += expr_str
+		}
 		if node.is_expr {
 			len := branch.stmts.len
 			if len > 0 {
@@ -303,8 +308,8 @@ fn (mut g Gen) comptime_if(node ast.IfExpr) {
 				g.writeln('}')
 			}
 		}
-		g.defer_ifdef = ''
 	}
+	g.defer_ifdef = ''
 	g.writeln('#endif')
 	if node.is_expr {
 		g.write('$line $tmp_var')

--- a/vlib/v/gen/c/fn.v
+++ b/vlib/v/gen/c/fn.v
@@ -296,6 +296,7 @@ fn (mut g Gen) gen_fn_decl(node &ast.FnDecl, skip bool) {
 			g.writeln('${fargtypes[i]}* ${fargs[i]} = HEAP(${fargtypes[i]}, _v_toheap_${fargs[i]});')
 		}
 	}
+	g.indent++
 	for defer_stmt in node.defer_stmts {
 		g.writeln('bool ${g.defer_flag_var(defer_stmt)} = false;')
 		for var in defer_stmt.defer_vars {
@@ -319,6 +320,7 @@ fn (mut g Gen) gen_fn_decl(node &ast.FnDecl, skip bool) {
 			}
 		}
 	}
+	g.indent--
 	if is_live_wrap {
 		// The live function just calls its implementation dual, while ensuring
 		// that the call is wrapped by the mutex lock & unlock calls.

--- a/vlib/v/tests/defer/defer_test.v
+++ b/vlib/v/tests/defer/defer_test.v
@@ -175,3 +175,17 @@ fn test_defer_with_reserved_words() {
 	eprintln('Done')
 	assert true
 }
+
+fn test_defer_inside_comptime_if_else() {
+	$if false {
+	} $else {
+		defer {
+		}
+	}
+	$if true {
+		defer {
+		}
+	} $else {
+	}
+	assert true
+}

